src/main/java/bobby/command/DeadlineCommand.java

```diff
@@ -0,0 +1,86 @@
+package bobby.command;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import bobby.Storage;
+import bobby.Ui;
+import bobby.exception.BobbyException;
+import bobby.exception.DeadlineException;
+import bobby.task.Deadline;
+import bobby.task.TaskList;
+
+/**
+ * Represents a 'deadline' command
+ */
+public class DeadlineCommand extends Command {
+    /** The full user input command */
+    private String fullCommand;
+    /** Specific date format that Bobby accepts as input */
+    private final SimpleDateFormat SIMPLE_DATE_FORMAT = new SimpleDateFormat("dd-MM-yyyy");
+
+    /**
+     * Creates a DeadLineCommand object.
+     *
+     * @param fullCommand User input command.
+     */
+    public DeadlineCommand(String fullCommand) {
+        this.fullCommand = fullCommand;
+    }
+
+    /**
+     * Checks whether the command includes a valid date.
+     *
+     * @param input The date to be inspected.
+     * @return True if the date is in the correct format.
+     */
+    public boolean isValidDate(String input) {
+        try {
+            SIMPLE_DATE_FORMAT.setLenient(false);
+            SIMPLE_DATE_FORMAT.parse(input);
+            return true;
+        } catch (ParseException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Carries out the respective command's actions.
+     *
+     * @param tasks TaskList object containing a list of Tasks.
+     * @param ui Ui object to allow for Bobby to print messages.
+     * @param storage Storage object that handles the reading/writing of TaskList into a specified file.
+     * @throws BobbyException if an invalid command is given by the user's input.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws BobbyException {
+        ui.printLongLine();
+        if (fullCommand.substring(8).isBlank()) { // nothing after command
+            throw new DeadlineException("blank");
+        } else if (!fullCommand.contains("/")) {  // no "/"
+            throw new DeadlineException("no_slash");
+        } else if (fullCommand.substring(fullCommand.indexOf("/") + 1).isBlank()) { // nothing after time
+            throw new DeadlineException("no_date");
+        } else if (!isValidDate(fullCommand.substring(fullCommand.length() - 10))) { // invalid date
```

Comment:
For long lines, perhaps including a line break after the operator could make it look neater 👍